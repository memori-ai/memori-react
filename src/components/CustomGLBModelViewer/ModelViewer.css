model-viewer {
    max-width: 100%;
    height: 400px;
    --poster-color: transparent;
}

@media (max-width: 800px) {
    model-viewer {
        height: 300px;
    }
}
