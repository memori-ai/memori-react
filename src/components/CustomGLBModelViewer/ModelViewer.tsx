import React, { useEffect } from 'react';

export interface Props {
  src: string;
  poster: string;
  alt?: string;
}

const ModelViewer = ({ src, poster, alt = '' }: Props) => {
  useEffect(() => {
    const script = document.createElement('script');
    script.src =
      'https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js';
    script.type = 'module';
    document.head.appendChild(script);
    return () => {
      document.head.removeChild(script);
    };
  }, []);
  return (
    <div className="model-viewer">
      {/* @ts-ignore */}
      <model-viewer
        src={src}
        ios-src=""
        poster={poster}
        alt={alt}
        shadow-intensity="1"
        disable-zoom
        camera-controls
        autoplay
      />
    </div>
  );
};

export default ModelViewer;
