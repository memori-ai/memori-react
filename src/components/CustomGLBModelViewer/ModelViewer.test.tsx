import React from 'react';
import { render } from '@testing-library/react';
import ModelViewer from './ModelViewer';

it('renders ModelViewer unchanged', () => {
  const { container } = render(
    <ModelViewer
      src="https://assets.memori.ai/api/v2/asset/b791f77c-1a94-4272-829e-eca82fcc62b7.glb"
      poster="https://assets.memori.ai/api/v2/asset/d8035229-08cf-42a7-a532-ab051df2603d.png"
    />
  );
  expect(container).toMatchSnapshot();
});
