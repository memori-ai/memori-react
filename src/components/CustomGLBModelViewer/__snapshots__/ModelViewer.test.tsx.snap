// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ModelViewer unchanged 1`] = `
<div>
  <div
    class="model-viewer"
  >
    <model-viewer
      alt=""
      autoplay="true"
      camera-controls="true"
      disable-zoom="true"
      ios-src=""
      poster="https://assets.memori.ai/api/v2/asset/d8035229-08cf-42a7-a532-ab051df2603d.png"
      shadow-intensity="1"
      src="https://assets.memori.ai/api/v2/asset/b791f77c-1a94-4272-829e-eca82fcc62b7.glb"
    />
  </div>
</div>
`;
